src/main/kotlin/io/snyk/plugin/services/SnykAnalyticsService.kt

```diff
@@ -5,34 +5,74 @@ import com.intellij.openapi.components.Service
 import com.intellij.openapi.components.service
 import com.intellij.openapi.diagnostic.Logger
 import com.intellij.openapi.diagnostic.logger
-import io.snyk.plugin.analytics.EventProperties
-import io.snyk.plugin.analytics.Segment
-import io.snyk.plugin.net.SnykApiClient
+import ly.iterative.itly.Environment
+import ly.iterative.itly.IterativelyOptions
+import ly.iterative.itly.Options
+import ly.iterative.itly.ValidationOptions
+import ly.iterative.itly.segment.SegmentPlugin
+import snyk.analytics.AnalysisIsReady
+import snyk.analytics.AnalysisIsTriggered
+import snyk.analytics.DestinationsOptions
+import snyk.analytics.Identify
+import snyk.analytics.IssueIsViewed
+import snyk.analytics.Itly
+import snyk.analytics.ItlyLogger
+import snyk.analytics.ProductSelectionIsViewed
+import snyk.analytics.WelcomeIsViewed
+import java.util.Properties
 
 @Service
 class SnykAnalyticsService : Disposable {
     private val log = logger<SnykAnalyticsService>()
-    private val segment = Segment
+
+    private var analyticsCollectionEnabled = true
+
+    private var anonymousId = ""
+    private var userId = ""
 
     init {
         val settings = service<SnykApplicationSettingsStateService>()
-        segment.setAnalyticsCollectionEnabled(settings.usageAnalyticsEnabled)
-        val userId = obtainUserId(settings.token)
-        segment.setUserId(userId)
+        analyticsCollectionEnabled = settings.usageAnalyticsEnabled
+        anonymousId = settings.userAnonymousId
+        userId = obtainUserId(settings.token)
+
+        val environment = loadIterativelyEnvironment()
+        log.warn("Initializing Iteratively integration for $environment...")
+        Itly.getInstance().load(
+            DestinationsOptions.builder()
+                .iteratively(
+                    IterativelyOptions.builder()
+                        .flushIntervalMs(10_000)
+                        .flushQueueSize(5)
```

Comment:
Any reason why it's 5 and not the default one?

Reply:
We had 5 in previous Segment integration.
I think having it as default (across all plugins too) is a "right" approach. Good catch!